packages/babel-core/src/transformation/plugin.js

```diff
@@ -77,6 +77,10 @@ export default class Plugin extends Store {
 
     this.maybeInherit(loc);
 
+    if (this.raw._c) {
+      throw new Error(`Plugin ${JSON.stringify(i)} specified ${JSON.stringify(loc)} appears to be a different plugin instance. Possible duplicate babel-core versions.`);
```

Comment:
Can we add a comment to say when this might happen? This will be pretty impenetrable otherwise.